### DISCUSSION TITLE
Simplified response to package registry

### DISCUSSION THREAD

**Discussion by rentalhost** [npm/feedback#831] [Registry]
There are currently two ways to obtain the packument of a given package via Registry NPM: complete and simplified.

Let's get real information:

https://registry.npmjs.org/typescript

The complete `typescript` data, for example, is currently 17.9 MB. If we consider gzip, the transferred data is 1.5 MB.

If we consider the simple version using the `Accept: application/vnd.npm.install-v1+json` header, which reduces the size to 6.0 MB. However, the content is not received in gzip (due to how Cloudflare uses the "accept" header), as also shown in #749. So it is worse to load the simple version than the full version if we take into account the transferred data. If it were in gzip, the transfer would be 1.2 MB (about 20% less, for this case).

However, it would be even better if we had an even slimmer version that could specify which fields are needed from the reported JSON (a GraphQL would be ideal, but something simpler would suffice).

Suggestion:

```
// https://registry.npmjs.org/typescript?keys=name,time

{
     "name": "typescript",
     "time": {
         "modified": "2022-12-20T07:14:33.302Z",
         "created": "2012-10-01T15:35:39.553Z",
         "0.8.0": "2012-10-01T15:35:41.062Z",
         "0.8.1": "2012-11-15T20:53:44.894Z",
         "0.8.1-1": "2012-12-04T23:35:48.083Z",
         // ...
         "5.0.0-dev.20221219": "2022-12-19T07:11:38.691Z",
         "5.0.0-dev.20221220": "2022-12-20T07:14:32.968Z"
     }
}
```

In my real case, I just need the versions, without any further information.

```
// https://registry.npmjs.org/typescript?keys=keyof:time
```

**Comment by arcanis** [comment]
It'd be nice indeed, especially since the header-generated content is frequently missing important fields.

**Comment by jasonwilliams** [comment]
Thanks @rentalhost for raising this, just to add some anecdotal support.

VSCode has a feature where you can hover over a package name in package.json and you get back the current version.
It can be very delayed in showing any version at all.

I did some investigation and it does seem the main cause is the large response payload which comes back, on slow networks it makes quite a difference.

VSCode [only needs](https://github.com/microsoft/vscode/blob/main/extensions/npm/src/features/packageJSONContribution.ts#L316-L320):
- description
- version (only cares about latest)
- homepage link (optional)

But the response received is around 7MB, here is an example:
https://registry.npmjs.org/%40types%2Fnode

Prev issue raised before: https://github.com/microsoft/vscode/issues/141163

Ideally we would need a small, cached and gzipped response with only those fields (but I understand there needs to either be a lowest common denominator or a graphql style response). I would love to see one that doesn’t have every single version though, as many queries don’t need that. This would reduce bandwidth and time a lot!
